### DISCUSSION TITLE
Does `VCPKG_TARGET_TRIPLET` affect the main project along side the dependencies.

### DISCUSSION THREAD

**Discussion by roastduck** [microsoft/vcpkg#38783] [Q&A]
Dear team,

I intend to build my project with a proprietary compiler, so I create my own triplet overlay, and configure my project with `VCPKG_TARGET_TRIPLET` and `VCPKG_OVERLAY_TRIPLETS`. All the dependencies build fine. However, the main project that depends on the dependencies is still building with the default compiler. Is it by design? How should I configure my compiler for both the dependencies and the main project?

**Comment by Neumann-A** [comment]
Set `VCPKG_HOST_TRIPLET`....

**Reply by roastduck** [reply]
What does "HOST" mean? I thought HOST to be the host toolchain in cross compiling (which means host should be x86 when compiling to arm from a x86 machine). Does "HOST" have a different meaning?

**Reply by Neumann-A** [reply]
host ist the build/host machine running the build. Also see https://learn.microsoft.com/en-us/vcpkg/users/buildsystems/cmake-integration#vcpkg_host_triplet

**Reply by roastduck** [reply]
I think it matches my understanding. For example, if I want to build for an arm machine but running the compiler on a x86 machine, then I need to set host triplet to x86 (or leave it as default) and set the target triplet to arm. Right? But now I only got my dependencies built for arm (from a x86 host), but my main project is incorrectly built for x86 (also from the x86 host).

**Reply by Neumann-A** [reply]
> but my main project is incorrectly built for x86 (also from the x86 host).

You need to setup your toolchain for your project correctly. All vcpkg settings are just for within vcpkg.

**Reply by roastduck** [reply]
> > but my main project is incorrectly built for x86 (also from the x86 host).
> 
> You need to setup your toolchain for your project correctly. All vcpkg settings are just for within vcpkg.

But I need to set `-DCMAKE_TOOLCHAIN_FILE=/path/to/vcpkg/scripts/buildsystems/vcpkg.cmake` to use `vcpkg`, how should I also set my own toolchain then?

**Reply by Neumann-A** [reply]
`-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=<your toolchain>`

**Reply by roastduck** [reply]
I have figured out that I need to set `VCPKG_CHAINLOAD_TOOLCHAIN_FILE` BOTH by `-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=` AND by setting it in my triplet overlay. Is that by design?

**Reply by Neumann-A** [reply]
Yes. Maybe the triplet variable should have had a different name but conceptually they are the same. 
The triplets determines stuff for inside vcpkg. What your project does is determined by you.
